### PR TITLE
add reset_ctx

### PR DESCRIPTION
**Pull request by piiswrong**

### CODE REVIEW

File: python/mxnet/gluon/parameter.py [hunk 1]
```diff
@@ -202,30 +165,96 @@ def _init_impl(self, data, ctx):
...
+            raise ValueError("Cannot reset context for Parameter %s because it "
+                             "has not been initialized."%self.name)
+
+
     def set_data(self, data):
         """Sets this parameter's value on all contexts to data."""
         assert self._data is not None, \
             "Parameter %s has not been initialized"%self.name
         for arr in self.list_data():
             arr[:] = data
```
Comment (szha) [inline python/mxnet/gluon/parameter.py]: Should it handle `_deferred_init`?

Comment (piiswrong) [inline python/mxnet/gluon/parameter.py]: no, must be initialized first

Comment (szha) [inline python/mxnet/gluon/parameter.py]: and None data?

File: tests/python/unittest/test_nn.py [hunk 2]
```diff
@@ -279,5 +303,6 @@ def test_flatten():
 
 
 if __name__ == '__main__':
-    import nose
-    nose.runmodule()
+    test_symbol_block()
+    #import nose
+    #nose.runmodule()
```
Comment (szha) [inline tests/python/unittest/test_nn.py]: fix

**Pull request merged**